core/editor/src/document/document_message_handler.rs

```diff
@@ -97,16 +99,116 @@ impl MessageHandler<DocumentMessage, ()> for DocumentMessageHandler {
 					.into(),
 				);
 			}
+			CloseActiveDocument => {
+				responses.push_back(FrontendMessage::ShowConfirmationBox.into());
+				/*
+				log::debug!("CLOSE ACTIVE");
+				self.documents.remove(self.active_document);
+				responses.push_back(FrontendMessage::CloseDocument { document_index: self.active_document }.into());
+				if self.documents.len() > 0 {
+					if self.active_document == self.documents.len() {
+						self.active_document -= 1;
+					}
+					let lp = self.active_document_mut().layer_panel(&[]).expect("Could not get panel for active doc");
+					responses.push_back(FrontendMessage::ExpandFolder { path: Vec::new(), children: lp }.into());
+					responses.push_back(FrontendMessage::SetActiveDocument { document_index: self.active_document }.into());
+					responses.push_back(
+						FrontendMessage::UpdateCanvas {
+							document: self.active_document_mut().document.render_root(),
+						}
+						.into(),
+					);
+				} else {
+					responses.push_back(DocumentMessage::NewDocument.into());
+				}*/
+			}
+			CloseDocument(id) => {
+				assert!(id < self.documents.len(), "Tried to select a document that was not initialized");
+				// Remove doc from the backend store. Use 'id' as FE tabs and BE documents will be in sync.
+				self.documents.remove(id);
+				responses.push_back(FrontendMessage::CloseDocument { document_index: id }.into());
+
+				if self.documents.len() > 0 {
+					// If we have atleast one doc remaining, just select the correct doc.
+					if id == self.active_document {
+						// Currently selected doc is being closed
+						if id == 0 {
+							// Move to the next doc
+							self.active_document = 0;
+						} else {
+							// Move to the prev doc
+							self.active_document = id - 1;
+						}
+						let lp = self.active_document_mut().layer_panel(&[]).expect("Could not get panel for active doc");
+						responses.push_back(FrontendMessage::ExpandFolder { path: Vec::new(), children: lp }.into());
+						responses.push_back(FrontendMessage::SetActiveDocument { document_index: self.active_document }.into());
+						responses.push_back(
+							FrontendMessage::UpdateCanvas {
+								document: self.active_document_mut().document.render_root(),
+							}
+							.into(),
+						);
+					} else {
+						// Currently not selected doc is being closed
+						if self.active_document >= id {
+							// Active doc will move one space to the left
+							self.active_document -= 1;
+							responses.push_back(FrontendMessage::SetActiveDocument { document_index: self.active_document }.into());
+						}
+					}
+				} else {
+					// Close last tab and create a new blank tab automatically
+					self.active_document = 0;
+					responses.push_back(DocumentMessage::NewDocument.into());
+				}
```

Comment:
I'm concerned about the three layers of `if/else` nesting. This whole section of conditional logic can be simplified and a level of depth can be removed like so:

```
if self.documents.len() == 0 {
    // The two lines from the bottom else clause of your current code
} else if id == self.active_document {
    // The seventeen lines from your inner if clause
} else if self.active_document >= id {
    // The three lines from your inner else clause surrounded by the nested if
```

Feel free to break up the `else`/`else if` blocks to include comments for the whole block, like so:
```
} else {

// ...can instead have comments like...

}
// A comment for the following else block
else {
```